admob/src/common/rewarded_ad.cc

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "admob/src/include/firebase/admob/rewarded_ad.h"
+
+#include "admob/src/common/admob_common.h"
+#include "admob/src/common/rewarded_ad_internal.h"
+#include "app/src/assert.h"
+#include "app/src/include/firebase/future.h"
+
+namespace firebase {
+namespace admob {
+
+const char kUninitializedError[] =
+    "Initialize() must be called before this method.";
+
+RewardedAd::RewardedAd() {
+  FIREBASE_ASSERT(admob::IsInitialized());
+  internal_ = internal::RewardedAdInternal::CreateInstance(this);
+
+  GetOrCreateCleanupNotifier()->RegisterObject(this, [](void* object) {
+    FIREBASE_ASSERT_MESSAGE(
```

Comment:
The idea of CleanupNotifier is to invalidate the objects and WARN the user they did the wrong thing, not assert.

Reply:
Noted. This was from the older implementation. Will fix it in banner and interstitial, too.

Reply:
Done.